fonts/generate.py

```diff
@@ -0,0 +1,657 @@
+    SVG version of a given font.
+
+    Base64-encodes the resulting WOFF2 font, and embeds it in a @font-face declaration.
+
+    :param opts: A set of options from the argument parser sub-command.
+    :return: True if successful, False otherwise.
+    """
+    fontname: str = opts.fontname
+    source_dir: str = opts.source
+    source_pth: Path = Path(source_dir)
+    font_data_pth: Path = Path(source_pth, fontname)
+    font_pth: Path = Path(font_data_pth, f"{fontname}.svg")
+    log.debug("Creating a subset SVG file from %s", font_pth.resolve())
+
+    supported_glyphs: dict = __combine_alternates_and_supported(opts)
+
+    log.debug("The resulting subset font will have %s glyphs", len(supported_glyphs.keys()))
+
+    Et.register_namespace("", SVG_NS["svg"])
+    svg_font = Et.parse(str(font_pth))
+    font_el: Optional[Et.Element] = svg_font.find(".//svg:defs/svg:font", SVG_NS)
+    if not font_el:
+        log.error("Could not find a font element in %s", font_pth.resolve())
+        return False
+
+    log.debug("Removing all hkern elements in since they are not needed.")
+    for hkern in font_el.findall(".//svg:hkern", SVG_NS):
+        font_el.remove(hkern)
+
+    for glyph in font_el.findall(".//svg:glyph", SVG_NS):
+        gname: Optional[str] = glyph.get("glyph-name")
+        if gname and gname[-4:] not in supported_glyphs:
+            font_el.remove(glyph)
+
+    log.debug("Shortening metadata entry to the essentials.")
+    metadata_el: Optional[Et.Element] = svg_font.find(".//svg:metadata", SVG_NS)
+    if metadata_el:
+        metadata_el.text = B64_FONT_LICENSE.format(fontname=fontname)
+
+    tmpdir = tempfile.mkdtemp()
+    log.debug("Created temporary directory %s", tmpdir)
+
+    tmp_svg_font = Path(tmpdir, f"{fontname}.svg")
+
+    log.debug("Writing SVG font %s", tmp_svg_font.resolve())
+    svg_font.write(str(tmp_svg_font), encoding="UTF-8", xml_declaration=True)
+
+    tmp_woff2_pth: Optional[Path] = __fontforge_svg2woff(opts, tmpdir)
+    if not tmp_woff2_pth:
+        log.error("Could not create WOFF2")
+        return False
+
+    css_filename: Path = Path(opts.data, f"{fontname}.css")
+    with open(Path(tmpdir, f"{fontname}.woff2"), "rb") as woff2_content:
+        b64_encoding: bytes = base64.urlsafe_b64encode(woff2_content.read())
+        with open(css_filename, "w") as css_content:
+            log.debug("Writing CSS file %s", css_filename.resolve())
+            fmt_css: str = FONTFACE_WRAPPER.format(
+                fontname=fontname, b64encoding=b64_encoding.decode()
+            )
+            css_content.write(fmt_css)
+
+    if opts.keep_intermediates:
+        intermediate_pth = Path(font_data_pth, "tmp")
+        log.debug("Keeping intermediate files in %s", intermediate_pth.resolve())
+        os.makedirs(intermediate_pth, exist_ok=True)
+        tmp_svg_font.replace(Path(intermediate_pth, f"{fontname}_subset.svg"))
```

Comment:
Again, `intermediate_pth` is as path object, so just use `(intermediate_pth / f"{fontname}_subset.svg")`

Reply:
I just can't bring myself to use a division operator on a string. Sorry! I know it's supported, but I'm not a fan. 😄